app/routes_tasks.py

```diff
@@ -0,0 +1,138 @@
+from flask import Blueprint, request, jsonify, make_response
+from app import db 
+from app.models.task import Task 
+from sqlalchemy import asc, desc
+from datetime import datetime
+import os, requests
+from dotenv import load_dotenv
+
+load_dotenv()
+
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+
+@tasks_bp.route("", methods=["GET"], strict_slashes=False)
+def get_tasks_or_tasks_by_title():
+    requested_title = request.args.get("sort")
+    
+    if requested_title == "asc": 
+        tasks = Task.query.order_by(Task.title.asc())
+    elif requested_title == "desc":
+        tasks = Task.query.order_by(Task.title.desc())
+    else: 
+        tasks = Task.query.all() 
+    
+    tasks_response = [] 
+
+    for task in tasks: 
+        tasks_response.append(task.tasks_to_json())
+
+    return jsonify(tasks_response), 200
+
+@tasks_bp.route("/<task_id>", methods=["GET"], strict_slashes=False)
+def get_one_task(task_id):
+    task = Task.query.get(task_id)
+    
+    if task: 
+        return jsonify(task.specific_task_to_json()), 200
+    
+    return make_response("", 404)
+
+@tasks_bp.route("", methods=["POST"], strict_slashes=False)
+def post_task(): 
+    """optional: 
+    Summary: Using a class method called Task.make_a_task to post a new task. 
+    
+    Parameter: none
+    
+    Results: response body with specific task information
+    """
+    request_body = request.get_json()
+    
+    keys = ["title", "description", "completed_at"]
+    for key in keys: 
+        if key not in request_body:
+            return {"details": "Invalid data"}, 400
+
+    new_task = Task.make_a_task(request_body, id=None)
+
+    db.session.add(new_task)
+    db.session.commit() 
+    return jsonify(new_task.specific_task_to_json()), 201
+
+@tasks_bp.route("<task_id>", methods=["PUT"], strict_slashes=False)
+def update_task(task_id): 
+    """optional 
+    Summary: Based on the make_a_task class method, update_task esesentially creates
+    a new task with the same task_id that is passed in from the route and the data to update, then update/replace the 
+    associated task row in the data base. 
+    
+    Parameter: task_id
+    
+    Returns: response body with specific task information
+    """
+    task = Task.query.get(task_id)
+    
+    if task: 
+        update_data = request.get_json() 
+
+        # update_data["task_id"] = task_id
+        # db.session.query(Task).update(update_data)
```

Comment:
It's good to get in the habit of removing commented out code before you start internship.